auth_mellon.h

```diff
@@ -85,6 +85,16 @@
 #define AM_ERROR_MISSING_PAOS_MEDIA_TYPE 3
 
 
+#ifdef WITH_DIAGNOSTICS
+typedef enum {
+    AM_DIAG_FLAG_ENABLED       = (1 << 0),
+    AM_DIAG_FLAG_LOG_SAML_REQ  = (1 << 1),
+    AM_DIAG_FLAG_LOG_SAML_RESP = (1 << 2),
```

Comment:
Yes, the intention is to use them later. The idea is you may want to trim down the volume of reported information to only what you're interested in. The idea was to set things up for this so it could be added later.

I agree with you the unreferenced symbols should be removed at this juncture. Done.